extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -200,20 +201,35 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
 
   /** @override */
   getAdUrl() {
-    // TODO: Check for required and allowed parameters. Probably use
-    // validateData, from 3p/3p/js, after moving it someplace common.
+    // TODO: Check for required parameters.
+    var data = {
+      'adClient' : this.element.getAttribute('data-ad-client'),
+      'adSlot' : this.element.getAttribute('data-ad-slot'),
+      'adHost' : this.element.getAttribute('data-ad-host'),
+      'adtest' : this.element.getAttribute('data-adtest'),
+      'tagOrigin' : this.element.getAttribute('data-tag-origin'),
+      'experimentId' : this.element.getAttribute('data-experiment-id'),
+      'ampSlotIndex' : this.element.getAttribute('data-amp-slot-index'),
+      'adChannel' : this.element.getAttribute('data-ad-channel'),
+      'autoFormat' : this.element.getAttribute('data-auto-format'),
+      'width' : Number(this.element.getAttribute('width')),
+      'height' : Number(this.element.getAttribute('height'))
+    };
+    validateAdSenseData(data);
```

Comment:
I don't think this'll do exactly the thing we want. data should probably originate from element.dataset (see previous comment) and also include any data attributes not specified in order for the validation to check that they aren't there.